Multisig/UI/Assets/CollectiblesViewController/FlatCollectiblesListViewModel.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  FlatCollectiblesListViewModel.swift
+//  Multisig
+//
+//  Created by Vitaly on 28.11.22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+struct FlatCollectiblesListViewModel {
+    var items: [CollectibleListItem] = []
+    var isEmpty: Bool {
+        items.isEmpty
+    }
+
+    var next: String?
+
+    init(_ items: [Collectible] = []) {
+        var collectibleListItems: [CollectibleListItem] = []
+        var currentNft: AddressString? = nil
+        items.forEach { collectible in
+            if currentNft != collectible.address {
+                collectibleListItems.append(CollectibleListItem.header(CollectibleHeader(name: collectible.tokenName ?? "Unknown", logoURL: URL(string: collectible.logoUri ?? ""))))
+                currentNft = collectible.address
+            }
+            collectibleListItems.append(CollectibleListItem.collectible(CollectibleItem(collectible: collectible)))
+        }
+        self.items = collectibleListItems
+    }
+
+    mutating func append(from list: Self) {
+        self.next = list.next
+        add(list.items)
+    }
+
+    mutating func add(_ models: [CollectibleListItem] = []) {
```

Comment:
What are the requirements for this method? Also, what does it guarantee to do?

Reply:
same as in FlatTransactionsListViewModel. It adds another page of items to the list.

Reply:
I understood that, but why is all the guard/let logic?

Reply:
does it expect the models content be in some configurations?

Reply:
header separators are inserted manually. If next page starts with items from group that started on the previous page, we have to make sure that first header is removed.

Reply:
Can you please add this to a method comment? Otherwise it is not clear what is the intent of the code, i.e. if it works as expected or not

On line 41 it uses "models[1]", is that correct? Should not it be "models[0]"?

Reply:
added comments, everything is correct as is. models[0] always contains header item at this point.

Reply:
maybe I would add explanation: 

```
// The models[0] always contains a header because it is a very first section on the page, and the header was added manually before.
```